### PR TITLE
Switch to raf+setTimeout from MC

### PR DESCRIPTION
**Pull request by cristianbote**
This is addressing #1341.

Basically, go with raf + setTimeout solution. There's a possible clamping here, but let's discuss it.

cc @andrewiggins @developit @marvinhagemeister 

~`-33B`~
`-30B`

### CODE REVIEW

File: hooks/src/index.js [hunk 1]
```diff
@@ -186,31 +186,25 @@ export function useContext(context) {
...
+	afterPaintEffects = [];
 }
 
 if (typeof window !== 'undefined') {
-	mc = new MessageChannel();
-
 	afterPaint = (component) => {
 		if (!component._afterPaintQueued && (component._afterPaintQueued = true) && afterPaintEffects.push(component) === 1) {
-			requestAnimationFrame(onPaint);
+			requestAnimationFrame(() => setTimeout(onPaint, 0));
```
Comment (marvinhagemeister) [inline hooks/src/index.js]: For a tiny bit more perf we could create this function outside of `afterPaint`.

Comment (cristianbote) [reply hooks/src/index.js]: Good point! Changed it. Thank you!

File: hooks/src/index.js [hunk 2]
```diff
@@ -186,31 +186,29 @@ export function useContext(context) {
  */
 let afterPaint = () => {};
 
-/** @type {MessageChannel} */
-let mc;
-
+/**
+ * Pending effects consumer.
+ */
 function onPaint() {
```
Comment (andrewiggins) [inline hooks/src/index.js]: Perhaps we can give this a more descriptive name? Something like `flushAfterPaintEffects()`

Comment (cristianbote) [reply hooks/src/index.js]: That's a fair point. That would give a bit more context on what it does. Renamed them.

**Pull request merged**